op-service/eth/errors.go

```diff
@@ -0,0 +1,22 @@
+package eth
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/ethereum/go-ethereum"
+)
+
+// MaybeAsNotFoundErr checks if the error is an ethereum.NotFound error
+// or has an error string that heuristically indicates that it is this error.
+// If so, it returns ethereum.NotFound, otherwise it returns the original error.
+func MaybeAsNotFoundErr(err error) error {
+	if errors.Is(err, ethereum.NotFound) ||
+		strings.Contains(err.Error(), "block not found") ||
+		strings.Contains(err.Error(), "header not found") ||
+		strings.Contains(err.Error(), "Unknown block") ||
+		strings.Contains(err.Error(), "unknown block") {
+		return ethereum.NotFound
+	}
```

Comment:
Can we have a hot-path for the following cases:
- `nil` to `nil`, to avoid `.Error` on a possible nil value, if `MaybeAsNotFoundErr` is (mis)used on a nil.
-  `errors.Is(err, ethereum.NotFound)` to catch the case where we don't call `err.Error()`

And then only call `err.Error()` once, and do `strings.ToLower()` on it to make the match case insensitive?

Reply:
I meant to separate it so that we can create the err string variable to not repeat the `err.Error()` stringification, but still do the `error.Is` check first (since that doesn't stringify)